src/v2/view-builder/views/ov/EnrollementChannelDataOktaVerifyView.js

```diff
@@ -1,29 +1,127 @@
+import CountryUtil from '../../../../util/CountryUtil';
+import SwitchEnrollChannelLinkView from './SwitchEnrollChannelLinkView';
 
 const Body = BaseForm.extend({
+  className: 'oie-enroll-ov-data',
   title () {
     return this.options.appState.get('currentAuthenticator').contextualData.selectedChannel === 'email' ?
       loc('oie.enroll.okta_verify.enroll.channel.email.title', 'login'):
       loc('oie.enroll.okta_verify.enroll.channel.sms.title', 'login');
   },
   save: loc('oie.enroll.okta_verify.setupLink', 'login'),
   getUISchema () {
-    const schemas = BaseForm.prototype.getUISchema.apply(this, arguments);
+    const uiSchemas = BaseForm.prototype.getUISchema.apply(this, arguments);
+    const phoneNumberUISchema = _.find(uiSchemas, ({ name }) => name === 'phoneNumber');
+    const phoneNumberUISchemaIndex = _.findIndex(uiSchemas, ({ name }) => name === 'phoneNumber');
+
+    const countryUISchema = {
+      'label-top': true,
+      label: loc('country.label', 'login'),
+      type: 'select',
+      options: CountryUtil.getCountries(),
+      name: 'country',
+      wide: true,
+    };
+
+    // Create an input group - serves as a display wrapper
+    const phoneNumberWithCodeUISchema = {
+      label: loc('mfa.phoneNumber.placeholder', 'login'),
+      type: 'group',
+      modelType: 'string',
+      'label-top': true,
+      name: 'phoneCode',
+      input: [
+        {
+          type: 'label',
+          label: `+${this.model.get('phoneCode')}`,
+          className: 'country-code-label',
+        },
+        Object.assign({}, phoneNumberUISchema),
+      ],
+    };
+
+    if (phoneNumberUISchemaIndex !== -1) {
+      // Replace phoneNumberUISchema..
+      uiSchemas.splice(phoneNumberUISchemaIndex, 1, phoneNumberWithCodeUISchema);
+      // Add countryUISchema before phone.
+      uiSchemas.splice(phoneNumberUISchemaIndex, 0, countryUISchema);
+    }
+
     const description = {
       View: this.options.appState.get('currentAuthenticator').contextualData.selectedChannel === 'email' ?
-        loc('oie.enroll.okta_verify.select.channel.email.description', 'login'):
-        loc('oie.enroll.okta_verify.select.channel.sms.description', 'login'),
+        loc('oie.enroll.okta_verify.channel.email.description', 'login'):
+        loc('oie.enroll.okta_verify.channel.sms.description', 'login'),
       selector: '.o-form-fieldset-container',
     };
-    schemas.push(description);
-    return schemas;
+
+    uiSchemas.push(description);
+    return uiSchemas;
+  },
+
+  handlePhoneCodeChange () {
+    const countryCodeField = this.el.querySelector('.country-code-label');
+    countryCodeField.innerText = `+${this.model.get('phoneCode')}`;
+  },
+
+  initialize () {
+    BaseForm.prototype.initialize.apply(this, arguments);
+    if (this.options.appState.get('currentAuthenticator').contextualData.selectedChannel === 'sms') {
+      this.listenTo(this.model, 'change:phoneCode', this.handlePhoneCodeChange.bind(this));
```

Comment:
Is there a modelEvents we could put this logic?

Reply:
We can but I wanted to selectively add the listener so I Added it here.